subgraphs/smolverse/src/mappings/erc721.ts

```diff
@@ -0,0 +1,38 @@
+import { log } from "@graphprotocol/graph-ts";
+
+import { ERC721, Transfer } from "../../generated/Smol Brains/ERC721";
+import { getIpfsJson, JSON } from "../helpers/json";
+import { updateTokenMetadata } from "../helpers/metadata";
+import { getOrCreateCollection, getOrCreateToken, getOrCreateUser } from "../helpers/models";
+import { isMint } from "../helpers/utils";
+
+// export function handleTransfer(event: Transfer, ipfsData: JSON | null = null): void {
+export function handleTransfer(event: Transfer): void {
+  const params = event.params;
+  const address = event.address;
+  const tokenId = params.tokenId;
+  const tokenIdString = tokenId.toString();
+
+  const collection = getOrCreateCollection(address);
+  const owner = getOrCreateUser(params.to.toHexString());
+  const token = getOrCreateToken(collection, tokenId);
+
+  const contract = ERC721.bind(address);
+  const tokenUri = contract.try_tokenURI(tokenId);
```

Comment:
Lets move this into the `isMint` if block. This is expensive so we only want to do it when needed.

Reply:
Fixed